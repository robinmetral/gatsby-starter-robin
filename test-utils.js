// wraps all tests with the emotion theme provider
// from https://testing-library.com/docs/react-testing-library/setup#custom-render

import React from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "emotion-theming"

import theme from "./src/theme"

const ThemeWrapper = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: ThemeWrapper, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
