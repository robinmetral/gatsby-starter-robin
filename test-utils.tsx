// wraps all tests with the emotion theme provider
// from https://testing-library.com/docs/react-testing-library/setup#custom-render

import React, { ReactNode, ReactElement } from "react"
import { render } from "@testing-library/react"
import { ThemeProvider } from "emotion-theming"
import { CacheProvider } from "@emotion/core"
import createCache from "@emotion/cache"

import theme from "./src/theme"
import GlobalStyles from "./src/components/GlobalStyles"

/* create emotion cache to avoid :first-child warnings */
const emotionCache = createCache()
emotionCache.compat = true

type TestWrapperProps = {
  children?: ReactNode
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  return (
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        {children}
      </ThemeProvider>
    </CacheProvider>
  )
}

const customRender = (ui: ReactElement, options?: object) =>
  render(ui, { wrapper: TestWrapper, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
