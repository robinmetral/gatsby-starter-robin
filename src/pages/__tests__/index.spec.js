import React from "react"
import { useStaticQuery } from "gatsby"
import { render } from "test-utils"
import IndexPage from "../index"

import indexStaticQuery from "./__fixtures__/indexStaticQuery"

beforeEach(() => {
  useStaticQuery.mockImplementation(() => indexStaticQuery)
})

describe("Index", () => {
  it("matches snapshot", () => {
    const index = render(<IndexPage />)
    expect(index).toMatchSnapshot()
  })
})
