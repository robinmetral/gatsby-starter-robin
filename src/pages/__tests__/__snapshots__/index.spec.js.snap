// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-4 {
  background: rebeccapurple;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  max-width: 300px;
  margin-bottom: 1.45rem;
}

<body>
    <div>
      <header
        class="emotion-4"
      >
        <div
          class="emotion-2 emotion-3"
        >
          <h1
            class="emotion-1"
            data-testid="title"
          >
            <a
              class="emotion-0"
              href="/"
            >
              Gatsby Default Starter
            </a>
          </h1>
        </div>
      </header>
      <div
        class="emotion-2 emotion-3"
      >
        <main>
          <h1>
            Hi people
          </h1>
          <p>
            Welcome to your new Gatsby site.
          </p>
          <p>
            Now go build something great.
          </p>
          <div
            class="emotion-5"
          >
            <div
              class=" gatsby-image-wrapper"
              style="position: relative; overflow: hidden;"
            >
              <div
                style="width: 100%; padding-bottom: 100%;"
              />
              <img
                alt=""
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAACYklEQVQ4y42Uy28SQRjA+dM8efDmwYN6qF6qiSY+Y/WgQRMibY00TaWNBSRSCraYQtHl/bR0KyxQWCgWWAqU8izl/Sq7rLNsRHlVJpvJtzPfb77nDIOcZHSoqZSrp4+KtXIziubaLRysMCZiCYqOoVnhjNEi8RcztdxxeTzc6VBfT+5O2Vhpb+vw4wMdZ0ppWvP9xzLeJoDNThf2W+Nz1+XzNxQubSToSKKW+BDc+WOnkshhSVgeCiGpViZMEg1oxc26Knt+ae3bEtJTZwzE1kXLccG0+sOOlrcvZXvsczPkITfsa20vwIKnhsh+BnjUarT74Gb13CY7KBVJMv3z4N1NszQYsMWM62HNrCis/GxXn0iYls23uz5LPBcv0bH8hUH2XRoM85ySXv7JBtO87jMIvWq+H5GoYIHCLA1ZxD6Qap3Ak8IKfW7TJ50lK6uP9E6RgndHaODtCJ6Z5RyHfnE7j6gRbcKlCYNSt+rtETHTpUGgEP8FYmdNqd/Mo7aiVWTfuH2L9xASvfxxlqr01EYkrJszvNkgW9bH0OuFr+99m+y9IOeyU6zIp/Hubp/yMEztlzFPwOhdvq+nIoS1JNn4t2sugCmVsDvPe2KKolnZLCxhOcAKQRDDXTQaVi46lqYhIBwHTrl3oWqhMRDtaJge37lOBMKo4tfbqhVX0J7snTsWps8uZWuoOQY6CcjpSIF55UvmqNgr5wUwtV1IVdnXtnSfPEB2qjDNqnvczRl0m+j6Jn5lXb6nAQJqinmN0ZEBj03YLzghY8PnTRz80o/GRJZpOLCb0PM9BN7pvUEjx28V00WUg9jIVwAAAABJRU5ErkJggg=="
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
              />
              <picture>
                <source
                  sizes="(max-width: 800px) 100vw, 800px"
                  srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w"
                />
                <img
                  alt=""
                  loading="lazy"
                  sizes="(max-width: 800px) 100vw, 800px"
                  src="/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png"
                  srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                />
              </picture>
              <noscript>
                &lt;picture&gt;&lt;source srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w" src="/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
              </noscript>
            </div>
          </div>
          <a
            href="/page-2/"
          >
            Go to page 2
          </a>
        </main>
        <footer>
          © 
          2019
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": .emotion-4 {
  background: rebeccapurple;
}

.emotion-2 {
  margin: 0 auto;
  max-width: 960px;
  padding: 1.45rem;
}

.emotion-1 {
  margin: 0;
}

.emotion-0 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 {
  max-width: 300px;
  margin-bottom: 1.45rem;
}

<div>
    <header
      class="emotion-4"
    >
      <div
        class="emotion-2 emotion-3"
      >
        <h1
          class="emotion-1"
          data-testid="title"
        >
          <a
            class="emotion-0"
            href="/"
          >
            Gatsby Default Starter
          </a>
        </h1>
      </div>
    </header>
    <div
      class="emotion-2 emotion-3"
    >
      <main>
        <h1>
          Hi people
        </h1>
        <p>
          Welcome to your new Gatsby site.
        </p>
        <p>
          Now go build something great.
        </p>
        <div
          class="emotion-5"
        >
          <div
            class=" gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              style="width: 100%; padding-bottom: 100%;"
            />
            <img
              alt=""
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAACYklEQVQ4y42Uy28SQRjA+dM8efDmwYN6qF6qiSY+Y/WgQRMibY00TaWNBSRSCraYQtHl/bR0KyxQWCgWWAqU8izl/Sq7rLNsRHlVJpvJtzPfb77nDIOcZHSoqZSrp4+KtXIziubaLRysMCZiCYqOoVnhjNEi8RcztdxxeTzc6VBfT+5O2Vhpb+vw4wMdZ0ppWvP9xzLeJoDNThf2W+Nz1+XzNxQubSToSKKW+BDc+WOnkshhSVgeCiGpViZMEg1oxc26Knt+ae3bEtJTZwzE1kXLccG0+sOOlrcvZXvsczPkITfsa20vwIKnhsh+BnjUarT74Gb13CY7KBVJMv3z4N1NszQYsMWM62HNrCis/GxXn0iYls23uz5LPBcv0bH8hUH2XRoM85ySXv7JBtO87jMIvWq+H5GoYIHCLA1ZxD6Qap3Ak8IKfW7TJ50lK6uP9E6RgndHaODtCJ6Z5RyHfnE7j6gRbcKlCYNSt+rtETHTpUGgEP8FYmdNqd/Mo7aiVWTfuH2L9xASvfxxlqr01EYkrJszvNkgW9bH0OuFr+99m+y9IOeyU6zIp/Hubp/yMEztlzFPwOhdvq+nIoS1JNn4t2sugCmVsDvPe2KKolnZLCxhOcAKQRDDXTQaVi46lqYhIBwHTrl3oWqhMRDtaJge37lOBMKo4tfbqhVX0J7snTsWps8uZWuoOQY6CcjpSIF55UvmqNgr5wUwtV1IVdnXtnSfPEB2qjDNqnvczRl0m+j6Jn5lXb6nAQJqinmN0ZEBj03YLzghY8PnTRz80o/GRJZpOLCb0PM9BN7pvUEjx28V00WUg9jIVwAAAABJRU5ErkJggg=="
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
            />
            <picture>
              <source
                sizes="(max-width: 800px) 100vw, 800px"
                srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w"
              />
              <img
                alt=""
                loading="lazy"
                sizes="(max-width: 800px) 100vw, 800px"
                src="/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png"
                srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w" sizes="(max-width: 800px) 100vw, 800px" /&gt;&lt;img loading="lazy" sizes="(max-width: 800px) 100vw, 800px" srcset="/static/6d91c86c0fde632ba4cd01062fd9ccfa/d1455/gatsby-astronaut.png 256w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/7e783/gatsby-astronaut.png 512w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png 800w" src="/static/6d91c86c0fde632ba4cd01062fd9ccfa/a54c6/gatsby-astronaut.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </div>
        <a
          href="/page-2/"
        >
          Go to page 2
        </a>
      </main>
      <footer>
        © 
        2019
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
